libraries/AP_OSD/AP_OSD_SITL.cpp

```diff
@@ -187,6 +181,28 @@ void *AP_OSD_SITL::update_thread_start(void *obj)
 // initialise backend
 bool AP_OSD_SITL::init(void)
 {
+    // initialise the window on the main thread (macOS compatibility)
+    load_font();
+    {
+        WITH_SEMAPHORE(AP::notify().sf_window_mutex);
+        w = new sf::RenderWindow(sf::VideoMode(video_cols*(char_width+char_spacing)*char_scale,
+                                               video_lines*(char_height+char_spacing)*char_scale),
+                                 "OSD");
+    }
+    if (!w) {
+        AP_HAL::panic("Unable to create OSD window");
+    }
+
+#if defined(__APPLE__)
+    // macOS only, poll the window to force display; run in worker for other OS.
+    sf::Event event;
+    while (w->pollEvent(event)) {
```

Comment:
how does this exit this loop and go onto the pthread_create() on macos?

Reply:
`pollEvent` is non-blocking and returns false if the event queue is empty (https://www.sfml-dev.org/documentation/2.5.1/classsf_1_1Window.php).